axelor-stock/src/main/java/com/axelor/apps/stock/service/PickedProductServiceImpl.java

```diff
@@ -0,0 +1,258 @@
+import com.axelor.apps.stock.db.TrackingNumber;
+import com.axelor.apps.stock.db.repo.MassStockMoveRepository;
+import com.axelor.apps.stock.db.repo.PickedProductRepository;
+import com.axelor.apps.stock.db.repo.StockLocationLineRepository;
+import com.axelor.apps.stock.db.repo.StockMoveLineRepository;
+import com.axelor.apps.stock.db.repo.StockMoveRepository;
+import com.axelor.apps.stock.db.repo.StoredProductRepository;
+import com.axelor.apps.stock.exception.StockExceptionMessage;
+import com.axelor.i18n.I18n;
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+
+public class PickedProductServiceImpl implements PickedProductService {
+  protected PickedProductRepository pickedProductRepository;
+  protected StoredProductRepository storedProductRepository;
+  protected MassStockMoveRepository massStockMoveRepository;
+  protected StockMoveRepository stockMoveRepository;
+  protected StockMoveLineRepository stockMoveLineRepository;
+  protected StockLocationLineRepository stockLocationLineRepository;
+  protected StockMoveService stockMoveService;
+  protected StockMoveLineService stockMoveLineService;
+
+  @Inject
+  public PickedProductServiceImpl(
+      PickedProductRepository pickedProductRepository,
+      StoredProductRepository storedProductRepository,
+      MassStockMoveRepository massStockMoveRepository,
+      StockMoveRepository stockMoveRepository,
+      StockMoveLineRepository stockMoveLineRepository,
+      StockLocationLineRepository stockLocationLineRepository,
+      StockMoveService stockMoveService,
+      StockMoveLineService stockMoveLineService) {
+    this.pickedProductRepository = pickedProductRepository;
+    this.storedProductRepository = storedProductRepository;
+    this.massStockMoveRepository = massStockMoveRepository;
+    this.stockMoveRepository = stockMoveRepository;
+    this.stockMoveLineRepository = stockMoveLineRepository;
+    this.stockLocationLineRepository = stockLocationLineRepository;
+    this.stockMoveService = stockMoveService;
+    this.stockMoveLineService = stockMoveLineService;
+  }
+
+  @Override
+  public PickedProduct createPickedProduct(
+      MassStockMove massStockMove,
+      Product product,
+      TrackingNumber trackingNumber,
+      StockLocation fromStockLocation,
+      BigDecimal currentQty,
+      BigDecimal pickedQty,
+      StockMoveLine stockMoveLine) {
+
+    PickedProduct pickedProduct = new PickedProduct();
+    pickedProduct.setPickedProduct(product);
+    pickedProduct.setTrackingNumber(trackingNumber);
+    pickedProduct.setFromStockLocation(fromStockLocation);
+    pickedProduct.setCurrentQty(currentQty);
+    pickedProduct.setUnit(product.getUnit());
+    pickedProduct.setPickedQty(pickedQty);
+    pickedProduct.setStockMoveLine(stockMoveLine);
+    pickedProduct.setMassStockMove(massStockMove);
+    return pickedProduct;
+  }
+
+  @Transactional
```

Comment:
`@Transactional(rollbackOn = {Exception.class})` for every Transactionnal (Except if we don't want to rollback on every exception)